botfront/imports/api/graphql/trackerStore/mongo/trackerStore.js

```diff
@@ -0,0 +1,185 @@
+import uuidv4 from 'uuid/v4';
+import Conversations from '../../conversations/conversations.model';
+import Activity from '../../activity/activity.model';
+import Projects from '../../project/project.model';
+import NLUModels from '../../nlu/nlu.model';
+
+export const getTracker = async (senderId, projectId, after, maxEvents = 100) => {
+    const aggregation = [
+        {
+            $match: {
+                _id: senderId,
+                projectId,
+            },
+        },
+        {
+            $project: {
+                // retreive the last elements of the array from the index after
+                // index - len give us the x last element we want to fetch
+                'tracker.sender_id': 1,
+                'tracker.slots': 1,
+                'tracker.latest_message': 1,
+                'tracker.latest_event_time': 1,
+                'tracker.followup_action': 1,
+                'tracker.latest_input_channel': 1,
+                'tracker.active_form': 1,
+                'tracker.latest_action_name': 1,
+                'tracker.events': { $slice: ['$tracker.events', { $subtract: [after, { $size: '$tracker.events' }] }] },
+                trackerLen: { $size: '$tracker.events' },
+            },
+        },
+        {
+            $project: {
+                'tracker.sender_id': 1,
```

Comment:
Use [$addFields](https://docs.mongodb.com/manual/reference/operator/aggregation/addFields/) instead of $project the second time to avoid repeating.